### PR TITLE
Add gotEquipment to AddPlayer

### PR DESCRIPTION
**Pull request by IsaacG**
Also update canInfect to support secret zombies

### CODE REVIEW

File: backend/api_calls.py [hunk 1]
```diff
@@ -358,17 +359,17 @@ def AddPlayer(request, firebase):
...
   results.append(firebase.put('/users/%s/players' % user, player, player_info))
 
   player_info = {
     'gameId': game,
     'userId': user,
-    'canInfect': start_as_zombie,
+    'canInfect': (request['startAsZombie'] or request['beSecretZombie']),
     'needGun' : request['needGun'],
-    'startAsZombie' : start_as_zombie,
+    'gotEquipment' : request['gotEquipment'],
```
Comment (DolceTriade) [inline backend/api_calls.py:372]: Do you need to add gotEquipment to required_args?

Comment (IsaacG) [reply backend/api_calls.py:372]: I'm trying to make all args required as much as possible. It's the client code filling them in so it's easier to just set them vs handling maybe-setting them and it's easier on the server vs dealing with missing keys or determining a default.

Comment (DolceTriade) [reply backend/api_calls.py:372]: But since you're not enforcing this, this can cause a KeyError.

Comment (IsaacG) [reply backend/api_calls.py:372]: Oh. You mean, "You need to add it to the required arg list" :)
Yes. Yes I do. Updating.

**Review by DolceTriade — APPROVED** [review]

**Pull request merged**